api/v1alpha3/talosconfig_webhook.go

```diff
@@ -31,8 +32,14 @@ func (r *TalosConfig) ValidateCreate() error {
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
-func (r *TalosConfig) ValidateUpdate(old runtime.Object) error {
-	return r.validate()
+func (r *TalosConfig) ValidateUpdate(oldRaw runtime.Object) error {
+	old := oldRaw.(*TalosConfig)
+
+	if !cmp.Equal(r.Spec, old.Spec) {
+		return apierrors.NewBadRequest("TalosConfig.Spec is immutable")
+	}
+
+	return nil
```

Comment:
dropping `r.validate()` here seems like a bug

Reply:
well, the current implementation of `validate()` makes it redundant as you can't really update the spec, so there's nothing to validate.
But if we add something else there in the future, it probably makes sense to keep it.

Reply:
that makes sense as well, I haven't though about it ;)